cores/esp32/esp32-hal-uart.c

```diff
@@ -118,6 +144,9 @@ void uartDisableInterrupt(uart_t* uart)
     uart->dev->int_ena.val = 0;
     uart->dev->int_clr.val = 0xffffffff;
 
+    // Free uart rx interrupt
+    _uart_interrupt_array[uart->num] = NULL;
```

Comment:
Setting this to NULL means that if there was an allocated structure, you will now leak it. You need to FREE it first.